blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -143,9 +162,7 @@ export default (
                 <Route path="pipeline" component={RunDetailsPipeline} onEnter={analytics.trackPipelineRunVisited}>
                     <Route path=":node" component={RunDetailsPipeline} />
                 </Route>
-                <Route path="changes" component={RunDetailsChanges} onEnter={analytics.trackPipelineRunChangesVisited} />
-                <Route path="tests" component={RunDetailsTests} onEnter={analytics.trackPipelineRunTestsVisited} />
-                <Route path="artifacts" component={RunDetailsArtifacts} onEnter={analytics.trackPipelineRunArtifactsVisited} />
+                <Route path=":action" getComponents={getRunDetailsAction} />
```

Comment:
Create a parameterized `Route` for run details actions since these can be contributed through the `jenkins-run.actions` extension point.